src/xkbcomp/include.c

```diff
@@ -267,11 +267,13 @@ FindFileInXkbPath(struct xkb_context *ctx, const char *name,
         if (file) {
             if (pathRtrn) {
                 *pathRtrn = buf;
-                buf = NULL;
+            } else {
+                free(buf);
             }
             *offset = i;
             goto out;
         }
+        free(buf);
```

Comment:
possibly other option:
```c
char buf[PATH_MAX];

sprintf(buf, ...)

if (everything is fine)
    pathRtrn = strdup(buf);
```

which means we don't need to track buf itself.

Reply:
Done. I introduced the new message `XKB_ERROR_INSUFFICIENT_BUFFER_SIZE`.